extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseQuickpick.ts

```diff
@@ -0,0 +1,180 @@
+	let dbs: string[] | undefined = undefined;
+
+	// Prompt 1b. Select database (if not already specified in connection profile)
+	let selectedDatabase: string;
+	if (connectionProfile.database && connectionProfile.database !== constants.master) {
+		selectedDatabase = connectionProfile.database;
+		connectionUri = await vscodeMssqlApi.connect(connectionProfile);
+	} else {
+		// Need to get list of databases from the server and prompt the user
+		connectionUri = await vscodeMssqlApi.connect(connectionProfile);
+		dbs = (await vscodeMssqlApi.listDatabases(connectionUri))
+			.filter(db => !constants.systemDbs.includes(db));
+
+		const dbSelection = await vscode.window.showQuickPick(
+			dbs,
+			{ title: constants.selectDatabase, ignoreFocusOut: true });
+		if (!dbSelection) {
+			// User cancelled
+			return undefined;
+		}
+		selectedDatabase = dbSelection;
+	}
+
+	// Prompt 2. Browse and Select existing project file - first show workspace projects, then browse option
+	let projectFilePath: string;
+
+	// Get workspace projects
+	const workspaceProjects = await getSqlProjectsInWorkspace();
+	const projectOptions: string[] = [constants.browseEllipsisWithIcon];
+
+	// Add workspace projects to the list
+	workspaceProjects.forEach(projectUri => {
+		projectOptions.push(projectUri.fsPath);
+	});
+
+	const projectSelection = await vscode.window.showQuickPick(
+		projectOptions,
+		{ title: constants.selectProjectFile, ignoreFocusOut: true });
+
+	if (!projectSelection) {
+		// User cancelled
+		return undefined;
+	}
+
+	if (projectSelection === constants.browseEllipsisWithIcon) {
+		// Show file browser
+		const projectFileUri = await vscode.window.showOpenDialog({
+			canSelectFiles: true,
+			canSelectFolders: false,
+			canSelectMany: false,
+			openLabel: constants.selectString,
+			title: constants.selectProjectFile,
+			filters: {
+				'SQL Projects': ['sqlproj']
+			}
+		});
+
+		if (!projectFileUri || projectFileUri.length === 0) {
+			// User cancelled
+			return undefined;
+		}
+
+		projectFilePath = projectFileUri[0].fsPath;
+	} else {
+		// User selected a workspace project
+		projectFilePath = projectSelection;
+	}
+
+	const project = await Project.openProject(projectFilePath);
+
+	//Prompt 3: Select the action
+	const actionSelection = await vscode.window.showQuickPick(
+		[constants.compareActionRadioButtonLabel, constants.updateActionRadioButtonLabel],
+		{
+			title: constants.actionLabel,
+			ignoreFocusOut: true
+		});
+	if (!actionSelection) {
+		// User cancelled
+		return undefined;
+	}
+
+	// Map the selected action to the enum
+	const selectedAction = actionSelection === constants.compareActionRadioButtonLabel
+		? UpdateProjectAction.Compare
+		: UpdateProjectAction.Update;
+
+	// Build the connectionDetails object expected by the schema-compare endpoints.
+	// This is a lightweight mapping that mirrors the shape used by the mssql/vscode-mssql APIs.
+	const connectionDetails = {
+		userName: connectionProfile.user,
```

Comment:
Let's add this back:

```
(connectionProfile as any).id
```

I think the STS connection management expects an ID, so there might be odd behavior or conflicts without one.